### DISCUSSION TITLE
How to mock stores from ariakit ?

### DISCUSSION THREAD

**Discussion by theo-mesnil** [ariakit/ariakit#2808] [Questions] [answered]
Hi, i try to mock stores from Drawer or Modal for example on react testing library, and i can't have the correct values to works.
How can i mock the stores ? 

Thanks.

**Comment by diegohaz** [comment]
Hey, can you give an example of how you're trying to do it? How the component and tests look like?

**Comment by theo-mesnil** [comment]
I mock like this 

```
export const MOCK_MODAL_STORE = {
  setOpen: jest.fn(),
  show: jest.fn(),
  hide: jest.fn(),
  toggle: jest.fn(),
  stopAnimation: jest.fn(),
  setContentElement: jest.fn(),
  setDisclosureElement: jest.fn(),
  getState: jest.fn(),
  setState: jest.fn(),
  useState: mockUseState,
}
```

and i pass this on:

```
test('should show menu items', () => {
    render(<UserDrawer store={MOCK_MODAL_STORE} />)

    const xxxx = screen.getByText('xxxxxx')
    expect(xxxx).toBeInTheDocument()
  })
```

On my component i pass the store.

If i do directly call the useDrawer from our librairy (who return useDialogStore)

```
test('should show menu items', () => {
    const drawer = useDrawer()
    render(<UserDrawer store={drawer} />)

    const xxxx = screen.getByText('xxxxxx')
    expect(xxxx).toBeInTheDocument()
  })
```

i've got 

![Capture d’écran 2023-09-12 à 10 35 03](https://github.com/ariakit/ariakit/assets/50322149/99762140-38ab-4e61-b0a4-3d2b3b3b80a2)

**Reply by diegohaz** [reply, marked as answer]
I think you're calling `useDrawer` outside of a React component?

Regardless, I wouldn't advise performing that type of mock, as it involves implementation details that could potentially change in future patches.

**Reply by theo-mesnil** [reply]
Ok, so is better to create a component with useDrawer for the tests ?

**Reply by diegohaz** [reply]
Yes, if that's how you expect users to consume your API.

**Reply by theo-mesnil** [reply]
Ok thanks this works :) Have a good day !
